### DISCUSSION TITLE
How to test Github actions without creating a release?

### DISCUSSION THREAD

**Discussion by dariogmori** [Arquisoft/faq#57] [Q&A]
I have recently created a new Dockerfile for compiling the API our team is coding in JavaSpring boot, which needs different containers for its Database and API. Because of this, I am in need of testing whether the Github actions are actually working properly or not, and I imagine I would not wanna do that on production (master). 
**Is there any way of testing the Github actions and deploying the application into a test environment without actually creating a release?**
Thanks.

**Comment by labra** [comment]
I am not sure to understand your question...I guess what you want to test are not the github actions but the docker images generated by the github actions, is that so?

**Reply by dariogmori** [reply]
The docker images work correctly, I have already tested them but I wanted to try and deploy with the Github secrets I defined and I was wondering if it was possible to do it in another environment other than production.

**Reply by Manueluz** [reply]
If all you want to do is rerun the deploy jobs over and over:

If you don't want to create a release every time (Assuming you haven't changed the .github files) you can go to Actions>Choose the latest deployment action in the list and press the Re-run All jobs to trigger a deploy without creating a new release.

![21a5ca9224a0f75692e7d67ff1d4dc43](https://github.com/Arquisoft/faq/assets/60448849/2011bca3-ae29-4b25-9bae-d46f269f7e05)

![090bea8ad564ab6a58a8297ab30e7152](https://github.com/Arquisoft/faq/assets/60448849/4c7976f8-cb9e-47c4-8d2f-8c96a5acc89b)

However, if you **have** changed the files inside .github folder you will need to create a new release.
